### PR TITLE
Add cancellation interop tests for Python

### LINKED ISSUE #1591: Implement cancellation interoperability tests in Python
Should be straightforward; we've supported cancellation from the beginning.

### PR DESCRIPTION
**Pull request by soltanmm**
Fixes #1591.

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
Why not also add this behavior to _interop_test_case.py?

**Comment by soltanmm** [comment]
Because I'm silly and forgetful of such things existing.

File: src/python/interop/interop/methods.py [hunk 1]
```diff
@@ -273,7 +289,34 @@ def _ping_pong(stub):
...
+            size=response_size),),
+        payload=messages_pb2.Payload(body=b'\x00' * payload_size))
+    pipe.add(request)
+    response = next(response_iterator)
+    # We test the contents of `response` in the Ping Pong test - don't check
+    # them here.
+    response_iterator.cancel()
+
+    errored = False
+    try:
```
Comment (nathanielmanistaatgoogle) [inline src/python/interop/interop/methods.py]: Doesn't

try:
  <expression that should raise>
except Exception:
  pass
else:
  <raise statement>

free you from having the "errored" local field?

Comment (soltanmm) [inline src/python/interop/interop/methods.py]: These kinds of things are some of the reasons why I'm not on the Python readability team. :-P

**Comment by nathanielmanistaatgoogle** [comment]
This is ready for merge, right? I don't see any outstanding review comments or anything else?

**Comment by soltanmm** [comment]
Yep.

**Pull request merged**